packages/postgress/src/MessagePickupRepositoryPg.ts

```diff
@@ -0,0 +1,616 @@
+      return 0
+    }
+  }
+
+  /**
+   * Adds a new message to the queue and processes it based on live session status.
+   *
+   * @param {AddMessageOptions} options - The options for adding a message.
+   * @param {string} options.connectionId - The ID of the connection.
+   * @param {string[]} options.recipientDids - Recipient DIDs for the message.
+   * @param {string} options.payload - The encrypted message payload.
+   * @returns { messageId, receivedAt:Promise<string> }- A promise resolving to the messageId and receivedAt of the added message.
+   * @throws {Error} Throws an error if the agent is not defined or if an error occurs during message insertion or processing.
+   */
+  public async addMessage(options: AddMessageOptions): Promise<string> {
+    const { connectionId, recipientDids, payload } = options
+    this.logger?.debug(`[addMessage] Initializing new message for connectionId: ${connectionId}`)
+
+    // Ensure the agent is defined
+    if (!this.agent) {
+      throw new Error('Agent is not defined')
+    }
+
+    try {
+      // Retrieve live session details for the given connection ID
+      const liveSession = await this.getLocalliveSession(connectionId)
+
+      // Insert the message into the database
+      const query = `
+      INSERT INTO queuedmessages(connectionid, recipientKeys, encryptedmessage, state) 
+      VALUES($1, $2, $3, $4) 
+      RETURNING id
+    `
+      const state = liveSession ? 'sending' : 'pending'
+      const result = await this.messagesCollection?.query(query, [connectionId, recipientDids, payload, state])
+
+      const messageId = result?.rows[0].id
+      const receivedAt = result?.rows[0].created_at
+      this.logger?.debug(`[addMessage] Message added with ID: ${messageId} for connectionId: ${connectionId}`)
+
+      // Process the message based on live session status
+      if (liveSession) {
+        this.logger?.debug(`[addMessage] Live session exists for connectionId: ${connectionId}`)
+        await this.agent.messagePickup.deliverMessages({
+          pickupSessionId: liveSession.id,
+          messages: [{ id: messageId, encryptedMessage: payload }],
+        })
+      } else {
+        // Verify if a live session exists on another instance
+        const verifyInstance = await this.getLiveSessionFromDB(connectionId)
+        this.logger?.debug(`[addMessage] Live session verification result: ${JSON.stringify(verifyInstance)}`)
+
+        if (!verifyInstance) {
+          // Send a push notification if no live session exists
+          const connectionRecord = await this.agent.connections.findById(connectionId)
+          const token = connectionRecord?.getTag('device_token') as string | null
+
+          if (token) {
+            this.logger?.debug(
+              `[addMessage] Sending notification for connectionId: ${connectionId} with token: ${token}`,
+            )
+            await this.pushNotification(token, messageId)
```

Comment:
All of this should be replaced by a call to the ConnectionInfo callback. Something like:

```
const connectionInfo = await this.connectionInfoCallback(connectionId)
if (connectionInfo?.pushNotificationCallback) {
  await connectionInfo.sendPushNotification(messageId)
}
```

Reply:
Example of usage from a DIDComm mediator:

```ts
const connectionInfoCallback = async (connectionId) => {
          const connectionRecord = await this.agent.connections.findById(connectionId)

          const token = connectionRecord?.getTag('device_token') as string | null
          const maxReceiveBytes = connectionRecord?.getTag('max_receive_bytes') as number | 0

          return {
   maxReceiveBytes,
   sendPushNotification: token ? (messageId) => { this.notificationSender.send(token, messageId }: undefined,
 }
}

await messagePickupRepository.initialize({ agent, connectionInfoCallback })
```

Reply:
Done!